packages/ui/src/components/Topic/EditTopicProperties/EditTopicProperties.tsx

```diff
@@ -28,6 +28,11 @@ export const EditTopicProperties: FunctionComponent<
   const millisecondsToTimeValue = millisecondsToTime(
     topic["retention.ms"].value
   );
+  const [initialPartitions] = useState<number>(topic.partitions.length);
+  const [retentionTimeValidated, setRetentionTimeValidated] =
+    useState<ValidatedOptions>(ValidatedOptions.default);
+  const [retentionSizeValidated, setRetentionSizeValidated] =
+    useState<ValidatedOptions>(ValidatedOptions.default);
```

Comment:
We can derive the validity at render time when we click save, but we also need to make sure that changing the value to a valid one removes the validation error by setting the state to default without having to click save. Could we/should we  also do that without using a state for this?

Reply:
as changing a value changes some state, since the fields are not [uncontrolled](https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components), yeah it will work